src/dependency-manager/src/manager.ts

```diff
@@ -130,21 +131,38 @@ export default abstract class DependencyManager {
    */
   protected async loadDatastores(parent_node: DependencyNode) {
     for (const [ds_name, ds_config] of Object.entries(parent_node.service_config.getDatastores())) {
-      const dep_node = new DatastoreNode({
+      let dep_node_config = {
         key: ds_name,
         service_config: parent_node.service_config,
-        image: ds_config.docker.image,
         tag: parent_node.tag,
-        ports: {
-          target: ds_config.docker.target_port,
-          expose: await this.getServicePort(),
-        },
         parameters: await this.getParamValues(
           parent_node.ref,
           ds_config.parameters,
           ds_name,
         ),
-      });
+      };
+      const environment_service_config = this.environment.getServices()[`${parent_node.name}:${parent_node.tag}`];
+      let dep_node;
+
+      if (environment_service_config?.datastores[ds_name]?.host) {
+        dep_node = new ExternalNode({
+          ...dep_node_config,
+          host: environment_service_config.datastores[ds_name].host!,
+          ports: {
+            target: ds_config.docker.target_port,
```

Comment:
Two comments right here:
1. The docker target_port is a perfect "default", but make sure to check the env config to see if the user specified a different port the external service is running on. That can be overridden just like the host can (a host is required in order to override the target_port though.
2. Specify the same port as both the target and expose port. Docker compose uses the target port to connect services together, but most systems will use the expose port to broker interprocess communication (e.g. Kubernetes services connect to peers using the ClusterIP port, not the process' port). While doing this, make sure to set the expose_port to required on the interface again.